wgpu-core/src/hub.rs

```diff
@@ -89,9 +95,10 @@ impl<T, I: TypedId> ops::Index<I> for Storage<T, I> {
     type Output = T;
```

Comment:
If I understand correctly, we are now going to be doing checked indexing at all times (or at least, mostly). This means, unfortunately, that we can't use `impl Index` any more, since we need to return a result...
We'll need something like
```rust
// returns Ok() for valid entries, Err(IdError) for Error entries, and panics for Vacant
fn get(&self, id: I) -> Result<Id, IdError> {
}
```